python/ray/__init__.py

```diff
@@ -257,5 +260,18 @@ def __getattr__(self, attr):
 ]
 
 
+if sys.version_info < (3, 7):
+    # TODO(Clark): Remove this one we drop Python 3.6 support.
+    from ray import autoscaler  # noqa: F401
+    from ray import data  # noqa: F401
+    from ray import workflow  # noqa: F401
+else:
+    # Delay importing of expensive, isolated subpackages.
+    def __getattr__(name: str):
+        if name in _subpackages:
+            return importlib.import_module("." + name, __name__)
```

Comment:
Note that, similar to your typical `from ray import data` statement, imports are cached, so this will translate to a dictionary lookup after the subpackage has been imported once.

Reply:
Is there a way to test it?

Reply:
Also if the import fails here, is the error message equivalent to when import <module> failed?

Reply:
I believe it should be equivalent, I can check.